Fabric.Identity.API/scripts/tests/Install-Identity-Utilities.tests.ps1

```diff
@@ -874,3 +874,194 @@ Describe 'Confirm-DiscoveryConfig Unit Tests'{
+                # Arrange
+                Mock -ModuleName Install-Identity-Utilities -CommandName Search-XMLChildNode -MockWith {$false}
+                $nodesToSearch = @("tenants","replyUrls","claimsIssuerTenant","allowedTenants","registeredApplications")
+
+                # Act
+                $runResult = Migrate-AADSettings -installConfigPath "install.config" -azureConfigPath "azuresettings.config" -nodesToSearch $nodesToSearch
+                
+                # Assert
+                $runResult | Should -Be $false
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Search-XMLChildNode -Scope It -Times 1 -Exactly
+            }
+            It 'Should succeed but not remove install.config settings if comparison files are different'{
+                # Arrange
+                $customObject = [PSCustomObject]@{
+                 Path = ".\"
+                 Owner = "TestOwner"
+                 Group = "TestUsers"
+                 AccessType = "Allow"
+                 Rights = "ModifyData"
+                 ObjectDifference = "<variable name = 'newitem' value = 'newvalue'/>"
+                }
+
+                Mock -ModuleName Install-Identity-Utilities -CommandName Search-XMLChildNode -MockWith {$true}
+
+                $nodesToSearch = @("tenants","replyUrls","claimsIssuerTenant","allowedTenants","registeredApplications","azureSecretName")
+                $xmlChildNodes = Get-XMLChildNodes -installConfigPath "install.config" -configSection "identity" -nodesToSearch $nodesToSearch -childNodeGetAttribute "name"
+
+                Mock -ModuleName Install-Identity-Utilities -CommandName Remove-XMLChildNodes -MockWith {}
+                Mock -ModuleName Install-Identity-Utilities -CommandName Add-XMLChildNodes -MockWith {}
+                Mock -ModuleName Install-Identity-Utilities -CommandName Compare-Object -MockWith {$customObject}
+
+                # Act
+                $runResult = Migrate-AADSettings -installConfigPath "install.config" -azureConfigPath "azuresettings.config" -nodesToSearch $nodesToSearch
+                
+                # Assert
+                $runResult | Should -Be $true
+                $xmlChildNodes.Count | Should -Be 6
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Search-XMLChildNode -Scope It -Times 1 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Remove-XMLChildNodes -Scope It -Times 2 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Add-XMLChildNodes -Scope It -Times 1 -Exactly
+                Assert-MockCalled -ModuleName Install-Identity-Utilities -CommandName Compare-Object -Scope It -Times 1 -Exactly
+            }
+        }
+    }
+}
+Describe 'Migrate-AADSettings' -Tag 'Integration'{
+    Context 'Migrating AAD Settings using Integration Tests'{
```

Comment:
Consider a few more test cases for edge cases - no config file found, no permissions to the config file, and malformed XML file.  Adding these can also help drive improved error messaged.

Reply:
I considered the no config file, but I believe I would have to do it outside of pester, but didn't have time yet to do that. Pester tests require functions vs just running a powershell script, because functions (isolated code) are easy to test. I think the other two, no permissions and malformed XML, are also good ideas. I will see what I can do.